File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/models/TransactionCategory.java
package com.androidcollider.easyfin.common.models;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author Ihor Bilous
 */

@AllArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode(exclude = {"name", "visibility"})
public class TransactionCategory {

    private int id;
    private String name;
    private int visibility;

    public TransactionCategory(int id, String name) {
        this(id, name, 1);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/models/TransactionCategory.kt
package com.androidcollider.easyfin.common.models

/**
 * @author Ihor Bilous
 */

class TransactionCategory(var id: Int, var name: String, var visibility: Int = 1) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TransactionCategory

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id
    }
}